### PR TITLE
Feature/add tests

### PR DESCRIPTION
**Pull request by vinkabuki**

### CODE REVIEW

File: src/torManager/torManager.test.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    options: {
+      env: {
+        LD_LIBRARY_PATH: libPath,
+        HOME: ZBAY_DIR_PATH
+      },
+      detached: true
+    }
+  })
+  await tor.init()
+  await tor.kill()
```
Comment (EmiM) [inline src/torManager/torManager.test.ts:26]: Shouldn't we spy on tor.process.kill (and maybe check if it succeeded) or at least assert that tor.kill didn't throw any error?

File: src/torManager/torManager.test.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+})
+
+test('start tor, do not kill tor process and start again', async () => {
+  const torPath = `${process.cwd()}/tor/tor`
+  const ports = await getPorts()
+  const libPath = `${process.cwd()}/tor`
+  const tor = new Tor({
```
Comment (EmiM) [inline src/torManager/torManager.test.ts:33]: You create and configure Tor object the same way in multiple tests. Maybe add a testUtils file, throw it in there as a function?

File: src/torManager/torManager.ts [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  torDataDirectory: string
+  torPidPath: string
+  socksPort: string
+  torPassword: string
+  torHashedPassword: string
+  constructor({ torPath, options, appDataPath, controlPort, socksPort }: IConstructor) {
+    this.torPath = torPath
+    this.options = options
+    this.services = new Map()
+    this.torControl = new TorControl({ port: controlPort, host: 'localhost', password: 'dupa' })
```
Comment (EmiM) [inline src/torManager/torManager.ts:40]: 'dupa'

File: src/tracker/tracker.ts [hunk 4]
```diff
@@ -38,8 +38,9 @@ export class Tracker {
       fs.mkdirSync(ZBAY_DIR_PATH)
     }
     const tor = new Tor({
-      torPath,
       appDataPath: ZBAY_DIR_PATH,
+      socksPort: 9152,
```
Comment (EmiM) [inline src/tracker/tracker.ts:42]: Maybe we'll use fp here as well?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,7 +1,26 @@
...
+### Added
+
+- Test for tor, connections manager, and dataserver
+- Secured tor control port with hashed password
+- Linter
+- method to close 
+
+### Changed
+
+- Refactored tor control port and tor manager.
```
Comment (EmiM) [inline CHANGELOG.md:14]: Add information that api changed addNewService -> createNewHiddenService

File: src/libp2p/connectionsManager.ts [hunk 6]
```diff
@@ -141,7 +143,7 @@ export class ConnectionsManager {
     this.createAgent()
 
     const listenAddrs = [`/dns4/${this.host}/tcp/${this.port}/ws`]
-    this.localAddress = `${listenAddrs}/p2p/${this.peerId.toB58String()}`
+    this.localAddress = `${listenAddrs[0]}/p2p/${this.peerId.toB58String()}`
```
[the remark this answers is not in the document]
Comment (vinkabuki) [reply src/libp2p/connectionsManager.ts:146]: linter, listenAddrs is one element array

File: src/libp2p/websocketOverTor.ts [hunk 7]
```diff
@@ -65,13 +70,13 @@ class WebsocketsOverTor extends WebSockets {
...
-  async _connect (ma, options: any = {}) {
-    if (options.signal && options.signal.aborted) {
+  async _connect(ma: multiaddr, options: any = {}) {
+    if (options.signal?.aborted) {
       throw new AbortError()
     }
     const cOpts = ma.toOptions()
     log('dialing %s:%s', cOpts.host, cOpts.port)
-    const myUri = `${toUri(ma)}/?remoteAddress=${encodeURIComponent(this.localAddress)}`
+    const myUri = `${toUri(ma) as string}/?remoteAddress=${encodeURIComponent(this.localAddress)}`
```
[the remark this answers is not in the document]
Comment (vinkabuki) [reply src/libp2p/websocketOverTor.ts:79]: toUri doesn't have typing so it returns 'any'

File: CHANGELOG.md (hunk 5 above)
Comment (EmiM) [reply CHANGELOG.md:14]: Btw. if API changes we should be bumping major version. We didn't do it but maybe it's a good time to start?

**Pull request merged**